packages/dds/cell/src/cell.ts

```diff
@@ -43,6 +43,42 @@ interface ICellValue {
     value: unknown;
 }
 
+// TODO: this should reference a shared interface in @fluidframework/runtime-definitions so it's usable from
+// here and @fluidframework/attributor
+interface AttributionInfo {
+    /**
+     * The user that performed the change.
+     */
+    user: string;
+    /**
+     * When the change happened.
+     */
+    timestamp: number;
+}
+
+/**
+ * {@inheritDoc CellAttributor}
+ */
+export class CellAttributor {
+
+    private info: AttributionInfo;
+
+    public constructor(initialInfo?: AttributionInfo) {
+        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+        this.info = initialInfo ?? {} as AttributionInfo;
+    }
+
+    public getAttributionInfo(): AttributionInfo {
+        return this.info;
+    }
+
+    public setAttributionInfo(message: ISequencedDocumentMessage): void {
+        this.info.user = message.clientId;
```

Comment:
note: `user` and `clientId` are different. The former is typically an `IUser` in code (which is an object containing an id, but also an extensible set of properties like "name", "email", etc.) whereas the latter is a guid.